nodejs_pipeline/run_swagger_generation.py

```
@@ -35,6 +35,155 @@ def should_process_directory(dir_path: str) -> bool:
+        }
+        api_index.setdefault(key, {"files": []})
+        _merge_file_entry(api_index[key]["files"], entry)
+    return api_index
+
+
+def _write_api_index(directory_path: str, api_index: dict) -> None:
+    output_path = _api_index_output_path(directory_path)
+    try:
+        with open(output_path, "w", encoding="utf-8") as f:
+            json.dump(api_index, f, indent=2)
+    except Exception:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)